plugins/out_oci_logan/oci_logan.c

```diff
@@ -0,0 +1,1327 @@
+                                 flb_sds_len(log_group_id));
+    }
+
+    if (!uri_param) {
+        return NULL;
+    }
+
+    // logSet
+    if (log_set) {
+        if (flb_sds_len(uri_param) > 0) {
+            uri_param = flb_sds_cat(uri_param, "&", 1);
+        }
+        uri_param = flb_sds_cat(uri_param, FLB_OCI_LOG_SET,
+                                 FLB_OCI_LOG_SET_SIZE);
+        uri_param = flb_sds_cat(uri_param, "=", 1);
+        uri_param = flb_sds_cat(uri_param, log_set,
+                                 flb_sds_len(log_set));
+    }
+
+    if (!uri_param) {
+        return NULL;
+    }
+
+    uri_param = flb_sds_cat(uri_param, "&", 1);
+    uri_param = flb_sds_cat(uri_param, FLB_OCI_PAYLOAD_TYPE,
+                            sizeof(FLB_OCI_PAYLOAD_TYPE) - 1);
+    uri_param = flb_sds_cat(uri_param, "=", 1);
+    uri_param = flb_sds_cat(uri_param, "JSON", 4);
+
+
+    if (!uri_param) {
+        return NULL;
+    }
+
+
+    if (flb_sds_len(uri_param) == 0) {
+        flb_sds_destroy(uri_param);
+        return flb_sds_create(ctx->uri);
+    }
+
+    full_uri = flb_sds_create_size(
+        flb_sds_len(ctx->uri) + 1 + flb_sds_len(uri_param));
+    if (!full_uri) {
+        flb_errno();
+        flb_sds_destroy(uri_param);
+        return NULL;
+    }
+
+    full_uri = flb_sds_cat(full_uri, ctx->uri, flb_sds_len(ctx->uri));
+    full_uri = flb_sds_cat(full_uri, "?", 1);
+    full_uri = flb_sds_cat(full_uri, uri_param, flb_sds_len(uri_param));
+
+    flb_sds_destroy(uri_param);
+
+    return full_uri;
+}
+
+static int flush_to_endpoint(struct flb_oci_logan *ctx,
+                             flb_sds_t payload,
+                             flb_sds_t log_group_id,
+                             flb_sds_t log_set_id)
+{
+    int out_ret = FLB_RETRY;
+    int http_ret;
+    size_t b_sent;
+    flb_sds_t full_uri;
+    struct flb_http_client *c = NULL;
+    struct flb_connection *u_conn;
+
+    full_uri = compose_uri(ctx, log_set_id, log_group_id);
+    if(!full_uri) {
+        flb_plg_error(ctx->ins, "unable to compose uri for logGroup: %s logSet: %s",
+                      ctx->oci_la_log_group_id, ctx->oci_la_log_set_id);
+    }
+
+    flb_plg_info(ctx->ins, "full_uri=%s", full_uri);
+
+    u_conn = flb_upstream_conn_get(ctx->u);
+    if(!u_conn) {
+        goto error_label;
+    }
+    /* Create HTTP client context */
+    c = flb_http_client(u_conn, FLB_HTTP_POST, full_uri, (void*) payload,
+                        flb_sds_len(payload), ctx->ins->host.name, ctx->ins->host.port, ctx->proxy, 0);
+    if (!c) {
+        goto error_label;
+    }
+    flb_http_allow_duplicated_headers(c, FLB_FALSE);
+
+    flb_plg_info(ctx->ins, "built client");
+    flb_http_buffer_size(c, FLB_HTTP_DATA_SIZE_MAX);
+    if (build_headers(c, ctx, payload, ctx->ins->host.name, ctx->ins->host.port, full_uri) < 0) {
+        flb_plg_error(ctx->ins, "failed to build headers");
+        goto error_label;
+    }
+    flb_plg_info(ctx->ins, "built request");
```

Comment:
it seems an info message its a bit verbose for the user, we usually get requirements around reducing the unnecessary logging, maybe you can do some flb_plg_debug()

Reply:
fixed